src/network/networkd-network.c

```diff
@@ -36,6 +36,51 @@
 #include "string-util.h"
 #include "util.h"
 
+static void network_config_hash_func(const void *p, struct siphash *state) {
+        const NetworkConfigSection *c = p;
+        unsigned filename_size = strlen(c->filename);
+
+        siphash24_compress(c->filename, filename_size, state);
+        siphash24_compress(&filename_size, sizeof(filename_size), state);
```

Comment:
there's no need to add the filename size explicitly to the hash. the filename itself should suffice